lib/Driver/Compilation.cpp

```diff
@@ -120,7 +126,8 @@ Compilation::Compilation(DiagnosticEngine &Diags,
 static bool writeFilelistIfNecessary(const Job *job, DiagnosticEngine &diags);
 
 using CommandSet = llvm::SmallPtrSet<const Job *, 16>;
-
+using CommandSetVec = llvm::SetVector<const Job*>;
+using BatchPartition = std::vector<std::unique_ptr<CommandSetVec>>;
```

Comment:
Why the indirection here?

Reply:
Hm I think this is residue from an iteration of the patch that paid more attention to partition-element identity. Will remove if so, good catch.